Let method and property docblocks inherit from parent classes and refactor.

hotoiledgoblinsack opened the pull request:
Hello

This further improves on your implementation of #34 by using the docblock from the method or property in parent classes if available, which is mostly useful for overrides so you don't have to copy the docblock over to get the description.

Peekmo commented:
Hello,

To be perfect, it should be good to parse ```{@inheritdoc}``` too.
Could you please do it before merging ? Or I could do it after if you want ;)

Regards,
Peekmo

hotoiledgoblinsack commented:
You are right, I forgot about that. It's been added. However, note that according to phpDocumentor, inheritDoc only works for the long description (not for the summary, which was the only one shown during auto completion). I've also done the following:

  * Refactored Tools.php so the docblock inheritance is now part of `getMethodArguments` and `getPropertyArguments` so other callers can also benefit from the inheritance.
  * Refactored the calls to `DocParser` in Tools.php so the docblock from the passed method or property is used, this avoids calling `get` up to four times (which creates about four `Reflector` instances per method/function/property). This will hopefully benefit performance in large projects.
  * The long description is now also shown in the tooltip under a separate heading.

hotoiledgoblinsack commented:
I can't commit right now, but I forgot to replace

```
returnValues = element.args.return.split('\\')
```

in **autocomplete-provider.coffee:68** with:

```
returnValues = if element.args.return then element.args.return.split('\\') else []
```

If I get home before you merge this, I'll try to get it in, if not, feel free to merge the pull request and modify this one line.
